onnxruntime/core/providers/cpu/nn/tfidfvectorizer.cc

```diff
@@ -145,8 +146,8 @@ struct TfIdfVectorizer::Impl {
     assert(ngram_id != 0);
     --ngram_id;
     assert(ngram_id < ngram_indexes_.size());
-    auto output_idx = static_cast<int64_t>(row_num) * output_size_ + ngram_indexes_[ngram_id];
-    assert(static_cast<size_t>(output_idx) < frequencies.size());
+    size_t output_idx = row_num * output_size_ + SafeInt<size_t>(ngram_indexes_[ngram_id]);
```

Comment:
general comment - in these expressions with multiple operations, I think we need to be careful about where to add SafeInt to make sure all operations are checked.
e.g., in this case, `SafeInt<size_t>(row_num) * output_size_ + ngram_indexes_[ngram_id]` would ensure that the multiplication is also checked.

https://github.com/dcleblanc/SafeInt/blob/d0178e3c3128b57041d6b0676028ef99c95b4f13/helpfile.md?plain=1#L545-L549